### ISSUE TITLE
Build for Microchip XC32 compiler fails due to RCON name conflict

### ISSUE THREAD

**Issue by kasjer** [Mbed-TLS/mbedtls#8338]
### Suggested enhancement

Static variable name change from **RCON** to something more descriptive.

### Justification

Mbed-TLS is used by mynewt project https://github.com/apache/mynewt-core.
It works fine with ARM and RISCV based compilers but is stopped to work with PIC32 XC32 compilers after some change in aes.c.

I guess eas.c includes assert.h which in turn for mynewt build also includes device specific headers and we have name conflict.

**aes.c** has static variable `RCON`

```c
static const uint32_t RCON[10] =
```
Unfortunately this directly conflicts with macro definition provided by PIC32 headers where RCON refers to some registers.

```txt
Error: In file included from c:\program files\microchip\xc32\v2.50\pic32mx\include\xc.h:689:0,
                 from repos/apache-mynewt-core/kernel/os/include/os/arch/pic32/os/os_arch.h:24,
                 from repos/apache-mynewt-core/kernel/os/include/os/os_fault.h:24,
                 from repos/apache-mynewt-core/libc/baselibc/include/assert.h:24,
                 from repos/mbedtls/library/common.h:32,
                 from repos/mbedtls/library/aes.c:26:
repos/mbedtls/library/aes.c:384:17: error: conflicting type qualifiers for 'RCON'
 static uint32_t RCON[10];
                 ^
c:\program files\microchip\xc32\v2.50\pic32mx\include\proc\pic32mz-ef\p32mz2048efh064.h:646:28: note: previous declaration of 'RCON' was here
 extern volatile uint32_t   RCON __attribute__((section("sfrs"), address(0xBF801240)));
```

I wander if Mbed-TLS project could change local static variable name to something else as this seems the easiest way to have crypto library work again for XC32 compiler.
This does not seem to impact library functionality but could make mynewt build work again.

Thanks

**Comment by ronald-cron-arm** [comment]
Thanks for the report. It seems the static RCON variable has been in aes.c for a long time thus I am not sure what change resulted in this new name conflict. You do not see it with mbedtls-3.4.1?

I think we could accept a PR that change the name of the static variable to a more descriptive one.

**Comment by tom-cosgrove-arm** [comment]
Agreed - `round_contants` or something like that

**Comment by kasjer** [comment]
Thanks for understanding, I will create PR with name change from `RCON` to `round_constants`.

FYI, this problem started in commit that added `#include <assert.h>` that for mynewt also adds some platform specific headers and for PIC32 included MCU related stuff.

**Comment by kasjer** [comment]
I could also make PRs to LTS branches especially to 2.28 since this is version that mynewt/nimble is currently based on.

**Comment by tom-cosgrove-arm** [comment]
Yup, that would be great - a backport to 2.28-LTS would be helpful alongside a main PR to `development`
